packages/remix-testing/create-remix-stub.tsx

```diff
@@ -41,98 +77,52 @@ type RemixStubOptions = {
   hydrationData?: HydrationState;
 
   /**
-   * The initial index in the history stack to render. This allows you to start a test at a specific entry.
-   * It defaults to the last entry in initialEntries.
-   * e.g.
-   *   initialEntries: ["/", "/events/123"]
-   *   initialIndex: 1 // start at "/events/123"
+   * Future flags mimicking the settings in remix.config.js
    */
-  initialIndex?: number;
-
   remixConfigFuture?: Partial<FutureConfig>;
-};
-
-function patchRoutesWithContext(
-  routes: (StubRouteObject | StubDataRouteObject)[],
-  context: AppLoadContext
-): (RouteObject | DataRouteObject)[] {
-  return routes.map((route) => {
-    if (route.loader) {
-      let loader = route.loader;
-      route.loader = (args) => loader({ ...args, context });
-    }
-
-    if (route.action) {
-      let action = route.action;
-      route.action = (args) => action({ ...args, context });
-    }
-
-    if (route.children) {
-      return {
-        ...route,
-        children: patchRoutesWithContext(route.children, context),
-      };
-    }
-
-    return route as RouteObject | DataRouteObject;
-  }) as (RouteObject | DataRouteObject)[];
 }
 
-interface StubIndexRouteObject
-  extends Omit<IndexRouteObject, "loader" | "action" | "children"> {
-  loader?: LoaderFunction;
-  action?: ActionFunction;
-  children?: StubRouteObject[];
-}
-
-interface StubNonIndexRouteObject
-  extends Omit<NonIndexRouteObject, "loader" | "action" | "children"> {
-  loader?: LoaderFunction;
-  action?: ActionFunction;
-  children?: StubRouteObject[];
-}
-
-type StubRouteObject = StubIndexRouteObject | StubNonIndexRouteObject;
-
-type StubDataRouteObject = StubRouteObject & {
-  children?: DataRouteObject[];
-  id: string;
-};
-
 export function createRemixStub(
-  routes: (StubRouteObject | StubDataRouteObject)[],
+  routes: StubRouteObject[],
   context: AppLoadContext = {}
 ) {
   return function RemixStub({
     initialEntries,
     initialIndex,
     hydrationData,
     remixConfigFuture,
-  }: RemixStubOptions) {
+  }: RemixStubProps) {
     let routerRef = React.useRef<Router>();
     let remixContextRef = React.useRef<RemixContextObject>();
 
     if (routerRef.current == null) {
-      // update the routes to include context in the loader/action
-      let patched = patchRoutesWithContext(routes, context);
+      remixContextRef.current = {
+        future: { ...remixConfigFuture },
+        manifest: {
+          routes: {},
+          entry: { imports: [], module: "" },
+          url: "",
+          version: "",
+        },
+        routeModules: {},
+      };
 
+      // Update the routes to include context in the loader/action and populate
+      // the manifest and routeModules during the walk
+      let patched = processRoutes(
+        // @ts-expect-error loader/action context types don't match :/
```

Comment:
Why don't they match?

Reply:
Remix forces a key/value context object in TS:

```ts
interface AppLoadContext {
  [key: string]: unknown;
}

interface DataFunctionArgs {
  request: Request;
  context: AppLoadContext;
  params: Params;
}
```

But React Router is more lenient since you can realistically pass anything through, it doesn't need to be key/value:

```ts
interface DataFunctionArgs {
  request: Request;
  params: Params;
  context?: any;
}
```